nextjs-algorooms/src/pages/profile/[profileUID]/index.tsx

```diff
@@ -23,6 +23,17 @@ export default ({
 
   const router = useRouter();
 
+  // Essentially filter out submissions with unique questionUID.
```

Comment:
As Zain said, this can be slow. Maybe try using a set instead of using `some`

```typescript

  const getUniqueSubmissions = () => {
    const submissionsSet = new Set();   // Set of questionUIDs
    const uniqueSubmissions = [];       // User's unique submissions

    // Go through each of the user's submissions
    // and only keep unique questions
    for (let submission of data.submissions) {
      if (!submissionsSet.has(submission.questionUID)) {
        submissionsSet.add(submission.questionUID);
        uniqueSubmissions.push(submission);
      }
    }

    return uniqueSubmissions;
  }
```